### PR TITLE
Popover Kit

### PR DESCRIPTION
**Pull request by jasperfurniss**
#### Breaking Changes

_Please reflect if your changes may break in some way (changes like removing/renaming props are examples of breaking changes)_

#### Checklist:

- [x] **SPECS** Please cover your changes with specs
- [x] **CHANGELOG** Please add an entry on `[Unreleased]` section to every functionality `ADDED`/`CHANGED`/`DEPRECATED`/`REMOVED`/`FIXED`

### CODE REVIEW

File: app/pb_kits/playbook/_playbook.scss [hunk 1]
```diff
@@ -59,3 +59,4 @@
 @import 'pb_toggle/toggle';
 @import 'pb_user/user';
 @import 'pb_user_badge/user_badge';
+@import 'pb_popover/popover';
```
Comment (jasoncypret) [inline app/pb_kits/playbook/_playbook.scss]: Alphabetical order

File: app/pb_kits/playbook/packs/examples.js [hunk 2]
```diff
@@ -133,3 +133,5 @@ WebpackerReact.setup(TitleDetail)
 WebpackerReact.setup(Toggle)
 WebpackerReact.setup(User)
 WebpackerReact.setup(UserBadge)
+import * as Popover from 'pb_popover/docs'
```
Comment (jasoncypret) [inline app/pb_kits/playbook/packs/examples.js]: Reformat to file structure

File: app/pb_kits/playbook/packs/kits.js [hunk 3]
```diff
@@ -57,3 +57,4 @@ import '../kits/pb_title_detail.js'
 import '../kits/pb_toggle.js'
 import '../kits/pb_user.js'
 import '../kits/pb_user_badge.js'
+import '../kits/pb_popover.js'
```
Comment (jasoncypret) [inline app/pb_kits/playbook/packs/kits.js]: Alphabetical order

File: app/pb_kits/playbook/pb_popover/_popover.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  ) {
+    this.button = button
+    this.tooltip = tooltip
+    this.placement = placement
+    this.offset = offset
+    this.setupPopper()
+  }
+
+  //getters
+  get popoverButton() {
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_popover/_popover.js]: What if this is used for tool tips or other pop-able elements. Thinking maybe this wouldn’t be a button. Possibly rename?

File: app/pb_kits/playbook/pb_popover/docs/_popover_list.html.erb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<div style="height: 750px; text-align: center;">
+<%= pb_rails("button", props: { text: "Button Secondary", variant: "secondary", id: 'list' }) %>
+<%= pb_rails("popover", props: {buttonId: "list", tooltipId: "list-tooltip", position: 'bottom', offset: false }) do %>
```
Comment (jasoncypret) [inline app/pb_kits/playbook/pb_popover/docs/_popover_list.html.erb]: ButtonID seems to limit this to only be buttons

Comment (jasoncypret) [reply app/pb_kits/playbook/pb_popover/docs/_popover_list.html.erb]: What is offset for?

File: app/pb_kits/playbook/pb_popover/_popover.js (hunk 4 above)
Comment (jasperfurniss) [reply app/pb_kits/playbook/pb_popover/_popover.js]: Any suggestions? I was thinking about 'Target' but that carries some possible meaning already in javascript... not sure.

File: app/pb_kits/playbook/pb_popover/docs/_popover_list.html.erb (hunk 5 above)
Comment (jasperfurniss) [reply app/pb_kits/playbook/pb_popover/docs/_popover_list.html.erb]: Not sure what we want to name it. ElementId? It's the element that is going to be clicked... definitely don't want the user to be pinned in by the terminology. Any other ideas? Basically the prop we need here is the ID of the "clickable trigger element". Trigger? TriggerId?

Comment (jasperfurniss) [reply app/pb_kits/playbook/pb_popover/docs/_popover_list.html.erb]: > What is offset for?

Offset provides the spacing between the clickable trigger element and the popover element. Seen here: https://popper.js.org/docs/v2/modifiers/offset/

File: app/pb_kits/playbook/pb_popover/_popover.js (hunk 4 above)
Comment (jasperfurniss) [reply app/pb_kits/playbook/pb_popover/_popover.js]: getpopoverTriggerElement() might be a little more generic for us here. 
It scales better-- no matter what the element is. Thoughts?

File: app/pb_kits/playbook/pb_popover/docs/_popover_list.html.erb (hunk 5 above)
Comment (jasperfurniss) [reply app/pb_kits/playbook/pb_popover/docs/_popover_list.html.erb]: > Not sure what we want to name it. ElementId? It's the element that is going to be clicked... definitely don't want the user to be pinned in by the terminology. Any other ideas? Basically the prop we need here is the ID of the "clickable trigger element". Trigger? TriggerId?

TriggerElementId

File: CHANGELOG.md [hunk 6]
```diff
@@ -6,6 +6,11 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 
+## [4.0.0] Unreleased
+
+###
+- Popover Kit (New Kit - Rails only) ([#569][] @jasperfurniss)
```
Comment (thestephenmarshall) [inline CHANGELOG.md]: You will probably want to rebase off `v4.0.0` to resolve conflicts

**Pull request merged**